pkg/live/load.go

```diff
@@ -369,6 +296,14 @@ func validateInventory(inventory kptfilev1.Inventory) error {
 
 func generateInventoryObj(inv kptfilev1.Inventory) *unstructured.Unstructured {
 	// Create and return ResourceGroup custom resource as inventory object.
+
+	// When inventoryID is not specified by the local resourcegroup, we generate one using
+	// depends-on annotation format that we store on the live cluster. This implementation detail
+	// is hidden from the local resourcegroup unless the one was explicitly generated by the client.
+	if inv.InventoryID == "" {
```

Comment:
I think the idea was that we would not implement any changes to the inventoryID in this PR? We know that generating an inventoryID this way is not safe.